packages/features/ee/workflows/components/WorkflowDetailsPage.tsx

```diff
@@ -96,20 +95,91 @@ export default function WorkflowDetailsPage(props: Props) {
       includeCalendarEvent: false,
       verifiedAt: SCANNING_WORKFLOW_STEPS ? null : new Date(),
       agentId: null,
+      inboundAgentId: null,
     };
     steps?.push(step);
     form.setValue("steps", steps);
   };
 
-  const agentQueriesTrpc = trpc.useQueries((t) =>
-    steps.map((step, index) => {
-      const watchedAgentId = form.watch(`steps.${index}.agentId`);
-      const agentId = step?.agentId ?? watchedAgentId ?? null;
+  // Watch all agent IDs at once instead of individually per step
+  const watchedSteps = useWatch({
+    control: form.control,
+    name: "steps",
+  });
+  console.log("watchedSteps", watchedSteps);
+  // Memoize agent IDs (both outbound and inbound) to prevent unnecessary query re-runs
+  const allAgentIds = useMemo(() => {
+    const outboundAgentIds = (watchedSteps || []).map((step) => step?.agentId || null);
+    const inboundAgentIds = (watchedSteps || []).map((step) => step?.inboundAgentId || null);
+    return { outboundAgentIds, inboundAgentIds };
+  }, [watchedSteps]);
+
+  // Only create queries for unique, non-null agent IDs (combining both types)
+  const uniqueAgentIds = useMemo((): string[] => {
+    const allIds = [...allAgentIds.outboundAgentIds, ...allAgentIds.inboundAgentIds];
+    const validIds: string[] = [];
+
+    for (const id of allIds) {
+      if (id && typeof id === "string") {
+        validIds.push(id);
+      }
+    }
+
+    return Array.from(new Set(validIds));
+  }, [allAgentIds]);
```

Comment:
This seems like it could be a hook e.g. useAgentIds ensuring uniqueness and other things